### PR TITLE
Redshift lineage crawler [sc-6671]

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

To add lineage support for Redshift.

### 🤓 What?

- Find table lineage from two system tables, `stl_scan` and `stl_insert`. [stl_scan](https://docs.aws.amazon.com/redshift/latest/dg/r_STL_SCAN.html) contains table scan steps for queries. [stl_insert](https://docs.aws.amazon.com/redshift/latest/dg/r_STL_INSERT.html) contains insert execution steps for queries. We can derive table lineage when a query is in both tables.
- Find view lineage from system catalog tables. [pg_depend](https://www.postgresql.org/docs/10/catalog-pg-depend.html) have table dependency.

### 🧪 Tested?

The local run connector successfully.

### CODE REVIEW

File: metaphor/redshift/lineage/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+This connector extracts lineage information from a Redshift database using [asyncpg](https://github.com/MagicStack/asyncpg) library.
+
+## Setup
+
+Follow the [Setup](../README.md#Setup) guide for the general Redshift connector to create the dedicated `metaphor` user. As the usage connector extracts information from system tables such as `STL_SCAN` & `STL_QUERY`, it needs to be given additional permissions using the following commands:
+
+```sql
+ALTER USER metaphor WITH SYSLOG ACCESS UNRESTRICTED;
+
+GRANT SELECT ON pg_catalog.svl_user_info TO metaphor;
```
Comment (elic-eon) [inline metaphor/redshift/lineage/README.md:12]: I will check the permission prerequisite deeply later

File: metaphor/redshift/lineage/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Redshift Lineage Connector
+
+This connector extracts lineage information from a Redshift database using [asyncpg](https://github.com/MagicStack/asyncpg) library.
+
+## Setup
+
+Follow the [Setup](../README.md#Setup) guide for the general Redshift connector to create the dedicated `metaphor` user. As the usage connector extracts information from system tables such as `STL_SCAN` & `STL_INSERT`, it needs to be given additional permissions using the following commands:
```
Comment (mars-lan) [inline metaphor/redshift/lineage/README.md:7]:
```suggestion
Follow the [Setup](../README.md#Setup) guide for the general Redshift connector to create the dedicated `metaphor` user. As the usage connector extracts information from system tables such as `STL_SCAN` & `STL_INSERT`, it needs to be given additional permissions using the following command:
```

File: tests/bigquery/usage/test_extractor.py [hunk 3]
```diff
@@ -36,5 +35,4 @@ async def test_extractor(test_root_dir):
 
         events = [EventUtil.trim_event(e) for e in await extractor.extract(config)]
 
-    print(json.dumps(events))
```
Comment (mars-lan) [inline tests/bigquery/usage/test_extractor.py:39]: Drop?

File: tests/redshift/lineage/test_extractor.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+from metaphor.common.base_config import OutputConfig
+from metaphor.common.event_util import EventUtil
+from metaphor.redshift.lineage.config import RedshiftLineageRunConfig
+from metaphor.redshift.lineage.extractor import RedshiftLineageExtractor
+from tests.test_utils import load_json
+
+
+# python3.7 do not have AsyncMock
+class MyAsyncMock(MagicMock):
```
Comment (mars-lan) [inline tests/redshift/lineage/test_extractor.py:15]: Name this `AsyncMock`?

File: tests/bigquery/usage/test_extractor.py (hunk 3 above)
Comment (elic-eon) [reply tests/bigquery/usage/test_extractor.py:39]: Yes, this line is removed.

Comment (mars-lan) [reply tests/bigquery/usage/test_extractor.py:39]: Oops sorry, misread it as added.

**Review by mars-lan — APPROVED** [review]

**Pull request merged**